sdk/core/core/src/az_http_pipeline.c

```diff
@@ -17,10 +17,19 @@ az_http_policy pipeline[] = {
   { .pfnc_process = NULL, .data = { 0 } },
 };
 
-AZ_NODISCARD az_result
-az_http_pipeline_process(az_http_request_builder * const hrb, const az_mut_span * const response) {
+AZ_NODISCARD az_result az_http_pipeline_process(
+    az_http_request_builder * const hrb,
+    const az_mut_span * const response,
+    az_http_policy_data * const policy_data) {
   AZ_CONTRACT_ARG_NOT_NULL(hrb);
   AZ_CONTRACT_ARG_NOT_NULL(response);
 
+  // FIXME global data
+  // Request auth instead of policy_data, and only set .credentials for the auth policy then?
+  // Not sure how this wassupposed to work, check with Rick.
+  for (size_t i = 0; i < sizeof(pipeline) / sizeof(pipeline[0]) - 1; ++i) {
```

Comment:
That was my question. I'll see what I can do.

Reply:
I talked to Jeffrey, it is a bit clearer now, but  I'll need to make some changes to the pipeline policies, such as make .data void* instead of az_span.